projects/sandbox/datagen/datagen/scripts/background.py

```diff
@@ -93,19 +93,37 @@ def validate_segments(
     ifos: List[str],
     sample_rate: float,
 ):
+
     segments = split_segments(segments, max_segment_length)
     validated = []
     for start, stop in segments:
-
+        duration = stop - start
         # using start/stops to decide if something
         # is a training file or not to make robust
         # to future use of multiple training background
-        is_train = train_start <= start <= (train_stop - minimum_train_length)
+        is_train = train_start <= start
+        is_train &= stop <= train_stop
+
         if is_train:
             subdir = "train"
             stop = min(stop, train_stop)
+            if duration < minimum_train_length:
```

Comment:
The one thing I would say is it might be worth checking this _after_ `validate_file`, since the reason we do checks here is that so downstream applications don't and can just load the data naively. So this is mostly a check as to whether we ought to download the data in the first place, but if the file for a bad segment exists I would think we want to flag it. That said, we obviously want to get this in today and with how long the `datagen` tests take, even if we think this should be changed maybe it's better to do in a later PR.

Reply:
I see. Agreed with all your points, let's get this in now and address later